### ISSUE TITLE
Changing languages requires a manual refresh to load new manifest data.

### ISSUE THREAD

**Issue by Daynar100** [joshhunt/destinySets#152]
Changing language using the language drop down does not cause it to load the new language. However, when the page is refreshed it does start loading the new language and there is a nice bar at the bottom with information about where it is in this process.

**Comment by Daynar100** [comment]
This is a very minor issue in comparison to anything that needs to get done with the release of forsaken. It's fine if I just need to figure it out on my own or if I should drop the issue. I started working on a solution to this and ran into a wall. In my [branch](https://github.com/Daynar100/destinySets/tree/language-change-trigger-manifest-reload) I got it to trigger the manifest reload by just reusing the code used to load the manifest in preloadStore and putting it in \Inventory\index.js fetchDefinitions function. I do plan on cleaning this up a bit, but that is not the issue.

The big issue is that there is a memory leak. The error 'Cannot enlarge memory arrays.' is thrown somewhere (not always after the same log, but I believe it is often in the getAllRecords call) in \lib\definitions\index.js allDataFromRemote function after switching between several languages. I've tried clearing the tables in the db variable, but the issue persists. I assume the old manifest data is still hanging around, but I'm not entirely sure how to properly manage memory in JS. Some guidance would be appreciated. @joshhunt This issue could probably be ignored as there isn't really a reason to change languages more than a once. It also doesn't break the site, the info bar on the bottom just gets stuck on whatever step it was on.

**Comment by joshhunt** [comment]
Your approach sounds fine, for now. Ideally it would be nice to have all of this in preloadStore (so I can move it to a more dedicated place later on), but having it in the `fetchDefinitions` in `Inventory/index.js` is fine.

I think that 'Cannot enlarge memory arrays.' error comes from the WebAssembly/asm.js sqlite library that's used to extract data out of the definitions data. You don't really need to worry about memory management in Javascript, the exception being when you're using C++ code compiled to run in Javascript.

I don't really have any pointers on how to deal with this tbh - I would have expected the sql library to handle this for us... @bhollis do you have any pointers on this one?

Does the error happen all the time, or only in certain browsers? One approach could be to just catch the error and reload the page... That will clear out the memory and things should work. While _far from perfect_, it gets the job done...

**Comment by bhollis** [comment]
It can totally be from using too much memory. I need to rebuild the wasm to see if newer Emscripten fixes that. What we do in DIM is force you to reload to get a new manifest, so we never load two at once.

**Comment by Daynar100** [comment]
It occurred in both browsers I tested. (Firefox and Edge) I messed around with some debug tools in Firefox and they seem to indicate that arrays/array buffers used in the sql library are the primary culprits. Forcing a refresh does 'fix' the issue. 

Instead of having it load a new manifest it could ask the user to refresh, or force a refresh whenever the language option is changed. This seems like a cop-out, but it is better than have language change appear to do nothing.
